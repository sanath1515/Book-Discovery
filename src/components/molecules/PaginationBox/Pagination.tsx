import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Pagination as MuiPaginationBox } from "@material-ui/lab";
import { COLORS } from "../../../theme/constants";

export type PaginationProps = {
	booksPerPage: number;
	totalBooks: number;
	getPage: (number: number) => any;
	page: number;
	onChange: (event: React.ChangeEvent<unknown>, page: number) => void;
};

const useStyles = makeStyles(theme => ({
	root: {
		border: "1px solid",
		borderColor: COLORS.GREY_300,
		borderRadius: "4px",
		width: "fit-content",
		"& .MuiPaginationItem-root": {
			margin: theme.spacing(0),
			border: "0",
		},
		"& .MuiPaginationItem-outlinedPrimary.Mui-selected": {
			backgroundColor: theme.palette.primary.main,
			color: "white",
		},
	},
}));

const Pagination: React.FC<PaginationProps> = ({
	booksPerPage,
	totalBooks,
	getPage,
	page,
	onChange,
}) => {
	var noOfPages = 0;

	for (let i = 1; i <= Math.ceil(totalBooks / booksPerPage); i++) {
		noOfPages++;
	}
	const classes = useStyles();
	console.log("Pagein", page);
	return (
		<MuiPaginationBox
			count={noOfPages}
			page={page}
			variant="outlined"
			shape="rounded"
			className={classes.root}
			color="primary"
			onChange={onChange}
		/>
	);
};

export default Pagination;
