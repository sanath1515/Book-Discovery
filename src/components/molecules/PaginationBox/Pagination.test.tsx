import { render } from "@testing-library/react";
import { ChangeEvent } from "react";
import Pagination from './Pagination';


describe('Pagination Testing', () => {

    it('Matches Snapshot', () => {
        const {container}=render(<Pagination booksPerPage={0} totalBooks={0} getPage={function (number: number) {
            throw new Error("Function not implemented.");
        } } page={0} onChange={function (event: ChangeEvent<unknown>, page: number): void {
            throw new Error("Function not implemented.");
        } } />)
        expect(container).toMatchSnapshot();
    })

    it('Pagination in the document',()=>{
        const {container}=render(<Pagination booksPerPage={0} totalBooks={0} getPage={function (number: number) {
            throw new Error("Function not implemented.");
        } } page={0} onChange={function (event: ChangeEvent<unknown>, page: number): void {
            throw new Error("Function not implemented.");
        } } />);
        expect(container).toBeInTheDocument();
    })
});