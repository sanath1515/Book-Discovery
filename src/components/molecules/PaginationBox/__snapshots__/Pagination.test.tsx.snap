// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Testing Matches Snapshot 1`] = `
<div>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root makeStyles-root-1"
  >
    <ul
      class="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
